odl/discr/diff_ops.py

```diff
@@ -815,102 +881,51 @@ def finite_diff(f, axis=0, dx=1.0, method='forward', out=None, **kwargs):
             raise ValueError('expected output shape {}, got {}'
                              ''.format(f.shape, out.shape))
 
-    if f_arr.shape[axis] == 2 and edge_order == 2:
-        raise ValueError('shape of array to small to use edge order 2')
+    if f_arr.shape[axis] == 2 and pad_mode == 'order2':
+        raise ValueError("shape of array to small to use 'order2'")
 
-    if pad_mode is not None:
-        if method == 'central' and edge_order == 1:
-            raise ValueError('central differences with padding cannot be used '
-                             'with first-order accurate edges')
-        if method in ('forward', 'backward') and edge_order == 2:
-            raise ValueError('{} differences with padding only use edge '
-                             'order 1'.format(method))
+    if kwargs:
+        raise ValueError('unkown keyword argument(s): {}'.format(kwargs))
 
     # create slice objects: initially all are [:, :, ..., :]
 
-    # current slice
-    slice_out = [slice(None)] * ndim
-
-    # slices used to calculate finite differences
-    slice_node1 = [slice(None)] * ndim
-    slice_node2 = [slice(None)] * ndim
-    slice_node3 = [slice(None)] * ndim
+    # Swap axes so that the axis of interest is first
```

Comment:
Yeah, I realized that when I read the code. A remark will make it clearer, though.